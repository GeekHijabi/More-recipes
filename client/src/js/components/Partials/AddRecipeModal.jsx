import React from 'react';
import Dropzone from 'react-dropzone';
import toastr from 'toastr';
import {
  Button, Modal,
  ModalHeader, ModalBody, Form, Label, Input, FormGroup, Col, FormText
} from 'reactstrap';
import imageUpload from '../../utils/imageUpload';

/**
 *
 *
 * @class AddRecipeModal
 * @extends {React.Component}
 */
class AddRecipeModal extends React.Component {
  /**
   * @description constructor
   * @param {any} props
   * @memberof Home
   * @return {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      recipeName: '',
      ingredients: '',
      description: '',
      imageUrl: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.onDrop = this.onDrop.bind(this);
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof AddRecipeModal
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof AddRecipeModal
 */
  onSubmit(event) {
    event.preventDefault();
    this.props.createRecipe(this.state).then(() => {
      toastr.options = {
        closeButton: true,
        progressBar: true
      };
      toastr.success('Recipe Created Successfully');
      this.props.toggle();
    });
  }

  /**
 * @returns {void}
 *
 * @param {any} files
 * @memberof AddRecipeModal
 */
  onDrop(files) {
    imageUpload(files)
      .then((response) => {
        const { body } = response;
        const fileURL = body.secure_url;
        this.setState({
          imageUrl: fileURL
        });
        console.log('our current image url', this.state.imageUrl);
      });
  }


  /**
   * @description constructor Function
   * @param {any} props
   * @memberof AddRecipeModal
   * @return {void}
   */
  render() {
    return (
      <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader toggle={this.props.toggle}>Edit recipe</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup row>
              <Label for="exampleEmail" sm={4}>Name</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="recipeName"
                  id="exampleEmail"
                  value={this.state.recipeName}
                  onChange={this.onChange}
                  placeholder="input Recipe name"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={4}>Ingredients</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="ingredients"
                  id="exampleEmail"
                  value={this.state.ingredients}
                  onChange={this.onChange}
                  placeholder="input your recipe ingredients"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={4}>description</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="description"
                  id="exampleEmail"
                  value={this.state.description}
                  onChange={this.onChange}
                  placeholder="Input a description for your recipe"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleFile" sm={4}>File</Label>
              <Col sm={8}>
                <Dropzone onDrop={this.onDrop} />
                <FormText color="muted" />
                {/* {this.state.imageUrl} */}
              </Col>
            </FormGroup>

            <FormGroup check row>
              <Col sm={{ size: 10, offset: 2 }}>
                <Button onClick={this.onSubmit}>
                  Add a recipe
                </Button>
              </Col>
            </FormGroup>
          </Form>
        </ModalBody>
      </Modal>
    );
  }
}

export default AddRecipeModal;
