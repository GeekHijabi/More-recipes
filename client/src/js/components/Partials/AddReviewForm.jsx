import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';

import { apiRecipeReview, onViewRecipe } from '../../actions/recipe';

/**
 *
 *
 * @class Banner
 * @extends {React.Component}
 */
class AddReviewForm extends React.Component {
/**
 *
 *@param {object} props
 * @returns {null} void
 * @memberof AddReviewForm
 */
  constructor(props) {
    super(props);
    this.state = {
      review: '',
    };
    this.onChange = this.onChange.bind(this);
    this.onClick = this.onClick.bind(this);
  }


//   /**
//  * @returns {void}
//  *
//  * @param {any} void
//  * @memberof AddReviewForm
//  */
//   componentWillMount() {
//     this.props.onViewRecipe(this.props.recipeId);
//   }


  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof SignIn
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }


  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof AddReviewForm
 */
  onClick(event) {
    const reviews = this.state.review;
    event.preventDefault();
    this.props.apiRecipeReview(this.props.recipeId, reviews);
  }

  /**
 *
 *
 * @class AddReviewForm
 * @extends {React.Component}
 */
  render() {
    return (

      <div className="review-form" placeholder="Add review">
        <div className="input-control">
          <label htmlFor="add-review" id="label">Post review</label>
          <input
            type="text"
            placeholder="input your review"
            id="add-review"
            name="review"
            onChange={this.onChange}
            value={this.state.review}
          />
        </div>
        <button
          type="button"
          className="btn blue-gradient btn-white btn-block btn-rounded z-depth-1a"
          onClick={this.onClick}
        >
             Add review
        </button>
      </div>
    );
  }
}
AddReviewForm.propTypes = {
  apiRecipeReview: PropTypes.func.isRequired,
  onViewRecipe: PropTypes.func.isRequired,
  recipeId: PropTypes.number.isRequired,
};

/**
 *
 * @param {object} state
 *
 * @returns {void}
 */
function mapStateToProps(state) {
  return {
    review: state.recipe.recipe.Reviews,
  };
}

export default connect(
  mapStateToProps,
  {
    apiRecipeReview,
    onViewRecipe
  }
)(AddReviewForm);
