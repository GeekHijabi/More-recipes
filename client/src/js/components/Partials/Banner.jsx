import React from 'react';
import PropTypes from 'prop-types';
import { withRouter } from 'react-router-dom';

const image = require('../../../assets/images/banner_bg.jpg');

/**
 *
 *
 * @class Banner
 * @extends {React.Component}
 */
class Banner extends React.Component {
  /**
   * @description COnstructor Function
   * @param {any} props
   * @param {any} context
   * @memberof Home
   * @return {void}
   */
  constructor(props) {
    super(props);
    this.state = {
    };
    this.onClick = this.onClick.bind(this);
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof Banner
 */
  onClick(event) {
    event.preventDefault();
    this.props.history.push('/recipes');
  }
  /**
   * @description COnstructor Function
   * @param {any} props
   * @memberof Home
   * @return {void}
   */
  render() {
    return (
      <div>
        <section id="banner" className="row">
          <img src={image} alt="" className="bg" />
          <div className="container">
            <div className="card card-body">
              <h1>Life is too short for just a rough meal</h1>
              <p className="btn">
                <button
                  type="button"
                  onClick={this.onClick}
                >
                Browse our collecton of Recipes
                </button>
              </p>
            </div>
          </div>
        </section>
      </div>
    );
  }
}

// Banner.contextTypes = {
//   router: React.PropTypes.func.isRequired,
// };

Banner.propTypes = {
  history: PropTypes.objectOf(PropTypes.any).isRequired
};


export default withRouter(Banner);
