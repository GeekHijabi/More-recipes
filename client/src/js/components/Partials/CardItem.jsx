import React from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';

const image = require('../../../assets/images/banner_bg.jpg');

/**
 *
 *
 * @class Banner
 * @extends {React.Component}
 */
class CardItem extends React.Component {
  /**
   * @description COnstructor Function
   * @param {any} props
   * @memberof Home
   * @return {void}
   */
  constructor(props) {
    super(props);
    this.state = {

    };
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof CardItem
 */

  /**
   * @description COnstructor Function
   * @param {any} props
   * @memberof Home
   * @return {void}
   */
  render() {
    const {
      recipe: {
        id,
        recipeName,
        imageUrl,
        upvotes,
        downvotes,
        favoriteCount
      }
    } = this.props;
    return (
      <div className="col-xs-12 col-sm-6 col-md-4 col-lg-3">
        <div className="card card-shape">
          <img
            className="card-img-top food-image"
            src={imageUrl || image}
            alt="Card cap"
            style={{ height: '180px' }}
          />
          <div className="card-body">
            <h4 className="card-title">{recipeName}</h4>
            <div className="row row-card">
              <div className="col-xs-12 col-sm-6 card-padding">
                <Link
                  className="btn card-view"
                  to={`/recipe/${id}`}
                  href={`/recipe/${id}`}
                >
                  View Details
                </Link>
              </div>
              <div className="col-xs-12 col-sm-6 card-paddings">
                <span className="ratings">
                  <i className="fa fa-thumbs-o-up 2x" />
                  <span>{upvotes || 0}</span>
                </span>
                <span className="ratings">
                  <i className="fa fa-thumbs-o-down 2x" />
                  <span>{downvotes || 0}</span>
                </span>
                <span className="ratings favRite">
                  <i className="fa fa-heart-o 2x" />
                  <span>{favoriteCount || 0}</span>
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}

CardItem.propTypes = {
  recipe: PropTypes.objectOf(String).isRequired,
};

export default CardItem;

