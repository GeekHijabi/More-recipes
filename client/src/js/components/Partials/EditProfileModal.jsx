import React from 'react';
import Dropzone from 'react-dropzone';
import toastr from 'toastr';
import {
  Button, Modal,
  ModalHeader, ModalBody, Form, Label, Input, FormGroup, Col, FormText
} from 'reactstrap';
import imageUpload from '../../utils/imageUpload';


/**
 *
 *
 * @class EditProfileModal
 * @extends {React.Component}
 */
class EditProfileModal extends React.Component {
  /**
   * @description Constructor Function
   * @param {any} props
   * @memberof EditProfileModal
   * @return {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      firstName: '',
      lastName: '',
      bio: '',
      summary: '',
      imageUrl: ''
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.onDrop = this.onDrop.bind(this);
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof EditProfileModal
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof EditProfileModal
 */
  onSubmit(event) {
    event.preventDefault();
    this.props.editProfile(this.state)
      .then(() => {
        toastr.options = {
          closeButton: true,
          progressBar: true
        };
        toastr.success('Profile edited Successfully');
        this.props.toggle();
      });
  }

  /**
 * @returns {void}
 *
 * @param {any} files
 * @memberof EditRecipeModal
 */
  onDrop(files) {
    imageUpload(files)
      .then((response) => {
        const { body } = response;
        const fileURL = body.secure_url;
        this.setState({
          imageUrl: fileURL
        });
        console.log('our current image url', this.state.imageUrl);
      });
  }


  /**
   * @description COnstructor Function
   * @param {any} props
   * @memberof EditProfileModal
   * @return {void}
   */
  render() {
    return (
      <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader toggle={this.props.toggle}>Edit Profile</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup row>
              <Label for="exampleEmail" sm={4}>firstName</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="firstName"
                  id="exampleEmail"
                  value={this.state.firstName}
                  onChange={this.onChange}
                  placeholder="input first name"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={4}>lastName</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="lastName"
                  id="exampleEmail"
                  value={this.state.lastName}
                  onChange={this.onChange}
                  placeholder="input last name"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={4}>bio</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="bio"
                  id="exampleEmail"
                  value={this.state.bio}
                  onChange={this.onChange}
                  placeholder="Input a bio describing you"
                />
              </Col>
            </FormGroup>
            <FormGroup row>
              <Label for="exampleEmail" sm={4}>summary</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="summary"
                  id="exampleEmail"
                  value={this.state.summary}
                  onChange={this.onChange}
                  placeholder="Input a cute summary of yourself"
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleFile" sm={4}>File</Label>
              <Col sm={8}>
                <Dropzone onDrop={this.onDrop} />
                <FormText color="muted" />
                {/* {this.state.imageUrl} */}
              </Col>
            </FormGroup>

            <FormGroup check row>
              <Col sm={{ size: 10, offset: 2 }}>
                <Button onClick={this.onSubmit}>
                   Edit Profile
                </Button>
              </Col>
            </FormGroup>
          </Form>
        </ModalBody>
      </Modal>
    );
  }
}

export default EditProfileModal;
