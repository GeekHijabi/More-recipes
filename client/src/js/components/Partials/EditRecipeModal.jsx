import React from 'react';
import toastr from 'toastr';
import {
  Button, Modal,
  ModalHeader, ModalBody, Form, Label, Input, FormGroup, Col
} from 'reactstrap';
import PropTypes from 'prop-types';
import { RingLoader } from 'react-spinners';
import imageUpload from '../../utils/imageUpload';

/**
 *
 *
 * @class EditRecipeModal
 * @extends {React.Component}
 */
class EditRecipeModal extends React.Component {
  /**
   * @description constructor Function
   * @param {any} props
   * @memberof EditRecipeModal
   * @return {void}
   */
  constructor(props) {
    super(props);
    const {
      recipeName,
      ingredients,
      description,
      imageUrl
    } = this.props.recipe;
    this.state = {
      recipeName,
      ingredients,
      description,
      imageUrl,
      isLoading: false,
    };
    this.onChange = this.onChange.bind(this);
    this.onSubmit = this.onSubmit.bind(this);
    this.onDrop = this.onDrop.bind(this);
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof EditRecipeModal
 */
  onChange(event) {
    this.setState({ [event.target.name]: event.target.value });
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof EditRecipeModal
 */
  onSubmit(event) {
    event.preventDefault();
    this.props.editRecipe(this.props.recipe.id, this.state).then(() => {
      toastr.options = {
        closeButton: true,
        progressBar: true
      };
      toastr.success('Recipe edited Successfully');
      this.props.toggle();
    });
  }

  /**
 * @returns {void}
 *
 * @param {any} files
 * @memberof EditRecipeModal
 */
  onDrop(files) {
    this.setState({
      isLoading: true
    });
    imageUpload(files)
      .then((response) => {
        const { body } = response;
        const fileURL = body.secure_url;
        if (fileURL) {
          this.setState({
            imageUrl: fileURL,
            isLoading: false
          });
        }
      }).catch(() => {
        'unsuccessful upload';
      });
  }

  /**
   * @description COnstructor Function
   * @param {any} props
   * @memberof EditRecipeModal
   * @return {void}
   */
  render() {
    const { imageUrl } = this.state;
    return (
      <Modal isOpen={this.props.isOpen} toggle={this.props.toggle}>
        <ModalHeader toggle={this.props.toggle}>Edit recipe</ModalHeader>
        <ModalBody>
          <Form>
            <FormGroup row>
              <Label for="exampleEmail" sm={3}>Recipe Name</Label>
              <Col sm={8}>
                <Input
                  type="text"
                  name="recipeName"
                  id="exampleEmail"
                  value={this.state.recipeName}
                  onChange={this.onChange}
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={3}>Ingredients</Label>
              <Col sm={8}>
                <textarea
                  type="text"
                  name="ingredients"
                  id="exampleEmail"
                  className="styledTextarea"
                  value={this.state.ingredients}
                  onChange={this.onChange}
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleEmail" sm={3}>Description</Label>
              <Col sm={8}>
                <textarea
                  type="text"
                  name="description"
                  id="exampleEmail"
                  className="styledTextarea"
                  value={this.state.description}
                  onChange={this.onChange}
                />
              </Col>
            </FormGroup>

            <FormGroup row>
              <Label for="exampleFile" lg={4}>Image</Label>
              <Col sm={8}>
                <input
                  type="file"
                  name="image"
                  onChange={this.onDrop}
                  accept="image/*"
                />
                { this.state.isLoading ?
                  <RingLoader
                    color="#B0C038"
                    loading={this.props.isLoadingRecipe}
                  /> :
                  <img
                    src={imageUrl}
                    alt="sample"
                    className="styledImage"
                  />}
              </Col>
            </FormGroup>

            <FormGroup check row className="modal-button">
              <Col sm={{ size: 10, offset: 5 }}>
                {this.state.isLoading ? 'please wait a few seconds...' :
                <Button onClick={this.onSubmit}>
               Edit
                </Button>
            }
              </Col>
            </FormGroup>
          </Form>
        </ModalBody>
      </Modal>
    );
  }
}

EditRecipeModal.defaultProps = {
  isLoadingRecipe: true
};

EditRecipeModal.propTypes = {
  editRecipe: PropTypes.func.isRequired,
  recipe: PropTypes.objectOf(PropTypes.any).isRequired,
  toggle: PropTypes.func.isRequired,
  isOpen: PropTypes.bool.isRequired,
  isLoadingRecipe: PropTypes.bool


};
export default EditRecipeModal;
