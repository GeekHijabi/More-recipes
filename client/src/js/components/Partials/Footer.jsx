import React from 'react';
import 'bootstrap/dist/css/bootstrap.css';
import '../../../styles/index.scss';

/**
 *
 *
 * @class Footer
 * @extends {React.Component}
 */
class Footer extends React.Component {
  /**
   * @description COnstructor Function
   * @param {any} props
   * @memberof Home
   * @return {void}
   */
  constructor(props) {
    super(props);
    this.state = {
    };
  }
  /**
   * @description COnstructor Function
   * @param {any} props
   * @memberof Home
   * @return {void}
   */
  render() {
    return (
      <footer className="page-footer">
        <div className="container">
          <div className="row">
            <div className="col-md-12">
              <div className="footer-socials mb-5 flex-center">
                <a className="icons-sm fb-ic" href="/">
                  <i className="fa fa-facebook fa-lg white-text mr-md-4" />
                </a>
                <a className="icons-sm tw-ic" href="/">
                  <i className="fa fa-twitter fa-lg white-text mr-md-4" />
                </a>
                <a className="icons-sm ins-ic" href="/">
                  <i className="fa fa-instagram fa-lg white-text mr-md-4" />
                </a>
                <a className="icons-sm pin-ic" href="/">
                  <i className="fa fa-pinterest fa-lg white-text" />
                </a>
              </div>
            </div>
          </div>
        </div>

        <div className="footer-copyright">
          <div className="container-fluid">
              Â© 2017 Copyright: Morerecipes.com
          </div>
        </div>
      </footer>
    );
  }
}

export default Footer;
