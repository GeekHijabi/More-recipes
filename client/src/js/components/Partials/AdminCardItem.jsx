import React from 'react';
import { Link } from 'react-router-dom';
import Proptypes from 'prop-types';
import { confirmAlert } from 'react-confirm-alert';
import 'react-confirm-alert/src/react-confirm-alert.css';
import EditRecipeModal from '../Partials/EditRecipeModal';

const defaultImage = require('../../../assets/images/no_image.png');

/**
 *
 *
 * @class Banner
 * @extends {React.Component}
 */
class AdminCardItem extends React.Component {
  /**
   * @description Constructor Function
   * @param {any} props
   * @memberof AdminCardItem
   * @return {void}
   */
  constructor(props) {
    super(props);
    this.state = {
      modal: false
    };
    this.onDelete = this.onDelete.bind(this);
    this.toggle = this.toggle.bind(this);
  }


  /**
   * @returns {void}
   *
   * @param {void} empty
   * @memberof AdminCardItem
   */
  onDelete() {
    confirmAlert({
      title: 'Confirm',
      message: 'Are you sure you want to do this.',
      confirmLabel: 'Confirm',
      cancelLabel: 'Cancel',
      onConfirm: () => this.props.deleteRecipe(this.props.recipe.id),
      onCancel: () => {},
    });
  }

  /**
 * @returns {void}
 *
 * @param {any} event
 * @memberof RecipeAdmin
 */
  toggle() {
    this.setState({
      modal: !this.state.modal
    });
  }

  /**
   * @description Render Method
   *
   * @memberof AdminCardItem
   * @return {dom} DomElement
   */
  render() {
    const {
      recipe: {
        id,
        recipeName,
        imageUrl
      }
    } = this.props;
    return (
      <div className="col-xs-12 col-sm-6 col-md-4 col-lg-3">
        <div className="card">
          <img
            className="card-img-top card-height"
            src={imageUrl || defaultImage}
            alt="Card cap"
          />
          <div className="card-body">
            <h4 className="card-title">{recipeName}</h4>
            <div className="row row-card">
              <div className="col-xs-12 col-sm-6 card-padding">
                <Link
                  className="btn card-view"
                  to={`/recipe/${id}`}
                  href={`/recipe/${id}`}
                >
                  View More
                </Link>
              </div>
              <div className="col-xs-12 col-sm-6 card-paddings">
                <span className="edit padding">
                  <i
                    role="button"
                    tabIndex="-1"
                    onClick={this.toggle}
                    data-toggle="tooltip"
                    title="Edit recipe"
                    onKeyPress={this.handleKeyPress}
                    className="fa fa-edit fa-2x fa-icons"
                  />
                </span>
                <EditRecipeModal
                  isOpen={this.state.modal}
                  toggle={this.toggle}
                  recipe={this.props.recipe}
                  editRecipe={this.props.editRecipe}
                />
                <span className="delete paddings">
                  <i
                    className="fa fa-trash fa-2x fa-icons"
                    role="button"
                    tabIndex="-1"
                    data-toggle="tooltip"
                    title="Delete profile"
                    onKeyPress={this.onKeyPress}
                    onClick={this.onDelete}
                  />
                </span>
              </div>
            </div>
          </div>
        </div>
      </div>
    );
  }
}
AdminCardItem.propTypes = {
  deleteRecipe: Proptypes.func.isRequired,
  editRecipe: Proptypes.func.isRequired,
  recipe: Proptypes.objectOf(Proptypes.any).isRequired
};

export default AdminCardItem;
