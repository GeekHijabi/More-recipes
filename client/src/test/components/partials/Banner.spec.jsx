import React from 'react';
import expect from 'expect';
import Adapter from 'enzyme-adapter-react-16';
import { shallow, configure } from 'enzyme';

import { Banner } from '../../../js/components/Partials/Banner';

configure({ adapter: new Adapter() });

const props = {
  history: {
    push: () => Promise.resolve({ ok: true }),
  }
};

describe('Banner', () => {
  it('should render correctly', () => {
    const wrapper = shallow(<Banner {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});

describe('onClick', () => {
  it('should click the link on the banner', () => {
    const event = {
      preventDefault: jest.fn()
    };
    const submit = shallow(<Banner {...props} />);
    submit.instance().onClick(event);
  });
});
