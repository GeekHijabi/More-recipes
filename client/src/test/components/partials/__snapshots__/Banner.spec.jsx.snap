// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Banner should render correctly 1`] = `
<section
  id="banner"
>
  <img
    alt=""
    className="bg"
    src={Object {}}
  />
  <div
    className="body-title-card"
  >
    <h1
      className="body-title"
    >
      Life is too short for just a rough meal
    </h1>
    <button
      className="body-title-button"
      onClick={[Function]}
      type="button"
    >
      Try out one of our awesome recipes
    </button>
  </div>
</section>
`;
